codegen/gateway.go

```diff
@@ -564,12 +563,33 @@ func NewEndpointSpec(
 	return augmentHTTPEndpointSpec(espec, endpointConfigObj, midSpecs)
 }
 
+func reloadTestFixtures(espec *EndpointSpec, endpointConfigObj map[string]interface{}) (*EndpointSpec, error) {
```

Comment:
This function seems to have a fat interface - It is not clear what this function does by just looking at the input/output of this func.

A better way is to define func `testFixtures(endpointConfigObj map[string]interface{}) (map[string]*EndpointTestFixture , error)`

Reply:
this follows the similar IO of `augmentHTTPEndpointSpec` below (and the same idea): augment http endpoint with generated tests.

EndpointSpec and endpointConfigObj seems to be the in-mem model (go struct) and its json representation at some extend. Test fixtures for example, it can be easily merged with json tag annotation, prefer to have them together and consistent with `augmentHTTPEndpointSpec`, what do you think?

Reply:
augmentHTTPEndpointSpec does lots of stuff for augmentations. So you can't know what is done by its signature but read the code. (It may not be a good thing but makes sense)

getting testFixture form config is a very specific task, and should have thin readable thin interface.

Reply:
thanks for the explanation, it makes sense, let's use `testFixtures(endpointConfigObj map[string]interface{}) (map[string]*EndpointTestFixture , error)`